sycl/test-e2e/KernelFusion/internalize_vfunc.cpp

```diff
@@ -48,11 +48,15 @@ int main() {
       cgh.parallel_for<class KernelOne>(numVec, [=](id<1> i) {
         size_t offset = i;
         vec<int, VEC> in1;
-        in1.load(offset, accIn1.get_pointer());
+        in1.load(offset,
+                 accIn1.template get_multi_ptr<access::decorated::legacy>());
```

Comment:
> What is the motivation for making this change? Does the old code not compile anymore? If so, that seems like an API break.

SYCL 2020 introduce this breakage (`get_pointer` now returns a raw pointer instead of the `multi_ptr`) and the `vec::load` has no raw pointer overload.

Reply:
However, there is an implicit conversion from raw pointer to the legacy `multi_ptr`:

```
// Legacy interface, inherited from 1.2.1.
// Deprecated.
template <typename ElementType, access::address_space Space>
class [[deprecated]] multi_ptr<ElementType, Space, access::decorated::legacy> {
 public:
  multi_ptr(ElementType*);
  /*...*/
};
```

I'm not 100% sure of the C++ rules here, but it seems like the compiler should allow passing a raw pointer to `vec::load` by going through the implicit conversion to legacy `multi_ptr`.  Since that legacy version is deprecated, I would expect a deprecation warning.

Reply:
> However, there is an implicit conversion from raw pointer to the legacy multi_ptr:

yes, and that's specifically what we wanted to ban with the new interface :smile:  (only safe with inferred address space as this is "equivalent" to an implicit cast from generic to concrete).

> I'm not 100% sure of the C++ rules here, but it seems like the compiler should allow passing a raw ...

You are right, but what is missing here is the template deduction. So `in1.load<access::address_space::generic_space, access::decorated::legacy>` should work. But IMHO, I think we should have had an overload taking raw pointers for `vec::load` and `vec::store`.

Reply:
> You are right, but what is missing here is the template deduction. So `in1.load<access::address_space::generic_space, access::decorated::legacy>` should work. But IMHO, I think we should have had an overload taking raw pointers for `vec::load` and `vec::store`.

Yes, I wasn't sure if the compiler would automatically deduce the template parameters after deciding to use the implicit conversion.  I guess you are saying that it will not, so code passing a raw pointer to `vec::load` will fail to compile.

It seems like we do not want to simply give a compilation failure for cases like this that used to work.  It would be better to compile successfully while giving a deprecation warning.  Adding raw pointer overloads for `vec::load` and `vec::store` could be reasonable if they are deprecated.  There must be other functions that take `multi_ptr` that have this same issue.  Should we add raw pointer overloads for those cases also?

Could we instead solve this by adding a deprecated constructor to the new multi_ptr that takes a raw pointer?